apps/src/lib/kits/maker/boards/microBit/MicroBitBoard.js

```diff
@@ -267,6 +267,18 @@ export default class MicroBitBoard extends EventEmitter {
       false /* shouldDestroyComponents */
     );
   }
+
+  calibrateCompass() {
+    return (
+      Promise.resolve()
+        .then(() => this.openSerialPort())
+        .then(serialPort => this.boardClient_.connectBoard(serialPort))
+        // Delay for 0.1 seconds to ensure time to establish port with micro:bit
```

Comment:
How comfortable are we that 0.1 seconds will be sufficient? If it is not, is it easy for the user to re-trigger the button or know when this error has occurred?